src/includes/parse.rs

```diff
@@ -30,6 +32,19 @@ use std::collections::HashMap;
 #[grammar = "includes/grammar.pest"]
 struct IncludeParser;
 
+/// Parses a single include block in the text
+///
+/// # Arguments
+/// The "start" argument is the index at which the include block starts.
+///
+/// It does not necessarily relate to the index of the include within the text str.
+///
+/// # Return values
+/// Returns a tuple of an [`IncludeRef`] that represents the included text and a usize that
+/// represents the end index of the include block.
+///
+/// This usize is not necessarily the end index within the text string passed in, and is
+/// solely based on the length of the include block and the start value.
```

Comment:
The `usize` is the offset at which to start scanning the next block. It is to enable iterative execution of the function to process includes.

Reply:
Yes, but the offset is not actually related to the string passed in, but instead to the string that is assumed to exist where this function is called from (line 91 of includes/mod.rs). This outer string is the actual full version of the string, unlike the passed version which is a slice and therefore should not be accessed by indices based on the full string. 

This implementation seems slightly strange, should I replace it with one that involves passing the full string, as well as a start index, as opposed to passing in a slice from that start index?

Reply:
I have changed the implementation of this function to explicitly use start in the slice, rather than implying its use in a slice outside the function. The documentation has been updated to show this new meaning.